pkg/controller/v1alpha1/inferencegraph/openshift_route_reconciler.go

```diff
@@ -104,6 +104,9 @@ func (r *OpenShiftRouteReconciler) buildOpenShiftRoute(inferenceGraph *v1alpha1.
 			Port: &v1.RoutePort{
 				TargetPort: intstr.FromString(inferenceGraph.GetName()),
 			},
+			TLS: &v1.TLSConfig{
+				Termination: v1.TLSTerminationReencrypt,
```

Comment:
I understand the reason for using Reencrypt, but if the customer changes the router's certificate, it seems like we would need to specify the CA certificate for the backend. Has that part already been applied elsewhere?

Reply:
Some banks have very strict security constraints and use SSL everywhere; sometimes, it is one of them.

But, as mentioned before, we can have an RFE to make SSL configurable.

Reply:
Using Reencrypted means that the certificate used by the Router is different from the one used by the backend. Therefore, it is common to provide the Router with a CA certificate that allows it to establish an SSL connection with the backend.

I’m not sure if this general Reencrypt concept applies differently to InferenceGraph, but if the backend is already set up with SSL, is there really a need for Reencrypted? Wouldn't passthrough be a better option?

Reply:
> but if the backend is already set up with SSL, is there really a need for Reencrypted? Wouldn't passthrough be a better option?

The back-end is setup with SSL with an _OpenShift service serving certificate_ which is only suitable for `*.cluster.local` hostnames. The back-end is always using this kind of certificate. AFAIK, the Router would always trust such certificates. Thus, `Reencrypt` is configured in the route to expose a proper certificate for the public Route.

Reply:
Sounds like the backend is oauth-proxy that uses *OpenShift service serving certificate* and Route(Reencrypt) does not need ca for `oauth-proxy`. 
I believe you tested this manually. If it works, I am ok it.

Reply:
I didn't exhaustive tests. I did only smoke tests and it was all working. I would hope the smokes are enough.